src/NodeManipulator/ToOneRelationPropertyTypeResolver.php

```diff
@@ -140,20 +146,13 @@ private function resolveFromObjectType(FullyQualifiedObjectType $fullyQualifiedO
             $types[] = new NullType();
         }
 
-        $propertyType = $this->typeFactory->createMixedPassedOrUnionType($types);
-
-        // add default null if missing
-        if (! TypeCombinator::containsNull($propertyType)) {
-            $propertyType = TypeCombinator::addNull($propertyType);
-        }
-
-        return $propertyType;
+        return $this->typeFactory->createMixedPassedOrUnionType($types);
     }
 
     private function isNullableType(?DoctrineAnnotationTagValueNode $joinDoctrineAnnotationTagValueNode): bool
     {
         if (! $joinDoctrineAnnotationTagValueNode instanceof DoctrineAnnotationTagValueNode) {
-            return false;
+            return true;
```

Comment:
default value is null

`\Doctrine\ORM\Mapping\JoinColumnProperties::$nullable` -> true